### PR TITLE
Bump rook version to 1.1.0-beta.0 and enable managed PDBs

### PR DESCRIPTION
**Pull request by rohantmp**
Haven't tried it out here. I'll comment when I have done a sanity check.

### CODE REVIEW

**Comment by kshlm** [comment]
/ok-to-test
/hold
Holding till we have a verification from @rohantmp

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,6 @@
 IMAGE_BUILD_CMD ?= "docker"
 REGISTRY_NAMESPACE ?= "ocs-dev"
+REGISTRY_NAME ?= "${REGISTRY_NAME}"
```
Comment (kshlm) [inline Makefile]: You need to set a default value here,
```make
REGISTRY_NAME ?= "quay.io"
```

File: cmd/manager/main.go [hunk 2]
```diff
@@ -25,7 +25,7 @@ import (
 	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
 	"sigs.k8s.io/controller-runtime/pkg/runtime/signals"
 
-	rookCephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
```
Comment (kshlm) [inline cmd/manager/main.go]: Please revert this rename. This is makes it hard to review the actual change adding poddisruptionbudget.

Comment (rohantmp) [reply cmd/manager/main.go]: Can I keep it in another commit?

File: Makefile (hunk 1 above)
Comment (rohantmp) [reply Makefile]: done, I had it like this in another branch, but used a vim replacement when moving the change to this branch. Thanks for catching it

File: cmd/manager/main.go (hunk 2 above)
Comment (rohantmp) [reply cmd/manager/main.go]: Currently it's in another commit.

**Comment by rohantmp** [comment]
@kshlm I think what's left is to generate a CSV with the rook image pointing at the 1.1 beta image

**Comment by rohantmp** [comment]
- The CephNFS CRD is not being pulled in causing problems when rook attempts to watch for it.
- The rook image needs to update to `rook/ceph:v1.1.0-beta.0`
- The OCSInitialization object is being created, but the cephfilesystem, cephobjectstore, and cephblockpools are not being initialized.

Adjusting for everything mentioned here, I have verified that everything is working with this patch.

**Comment by kshlm** [comment]
> * The CephNFS CRD is not being pulled in causing problems when rook attempts to watch for it. @davidvossel

Right now the only issue with this is that the logs get noisy. Apart from that, we don't have other issues.

> * The rook image needs to update to `rook/ceph:v1.1.0-beta.0`
> * The OCSInitialization object is being created, but the cephfilesystem, cephobjectstore, and cephblockpools are not being initialized.

OCSInitialization does nothing. All of this is to be handled by StorageClusterInitialization. This isn't being created automatically now, but will be.

File: cmd/manager/main.go (hunk 2 above)
Comment (kshlm) [reply cmd/manager/main.go]: Still, I don't see the need to rename. Unless this is to make the package name consistent to elsewhere in the code.

File: Makefile (hunk 1 above)
Comment (kshlm) [reply Makefile]: Still see the same here. This hasn't changed.

File: cmd/manager/main.go (hunk 2 above)
Comment (rohantmp) [reply cmd/manager/main.go]: It's to keep it consistent with everywhere else this package is used.

**Comment by rohantmp** [comment]
> The rook image needs to update to rook/ceph:v1.1.0-beta.0
@kshlm any updates on this? Should I generate the CSV or generate 0.0.2?

**Comment by rohantmp** [comment]
Fixed the Makefile issue. Had commited in another branch.

**Comment by kshlm** [comment]
> @kshlm any updates on this? Should I regenerate the current CSV or generate 0.0.2?

Just regenerate the CSV with the current version. Is there a 1.1 beta image of rook available? If there is, regenerate the CSV and submit it in this PR.

**Comment by davidvossel** [comment]
> The CephNFS CRD is not being pulled in causing problems when rook attempts to watch for it. @davidvossel

if ocs-operator doesn't support cephnfs, then we probably need rook to provide a flag to disable cephnfs entirely.

**Comment by rohantmp** [comment]
The failing CI seems to be a missing Make target @davidvossel ?

**Comment by kshlm** [comment]
CI is good. Your CSV doesn't seem to have been updated properly.

**Comment by davidvossel** [comment]
> The failing CI seems to be a missing Make target @davidvossel ?

"uncommitted CSV changes. run 'make gen-latest-csv' and commit results."

Some files are missing from the CSV generation made in this pr.

**Comment by kshlm** [comment]
You also need to commit the `hack/latest-csv-checksum.md5` file.

**Comment by rohantmp** [comment]
Gotta love that CI! Didn't know about `gen-latest-csv`

**Comment by rohantmp** [comment]
Gotta love that CI! Didn't know about `gen-latest-csv`.

**Comment by davidvossel** [comment]
e2e test fails because of a merge issue that's preventing the build compiling. 

```pkg/controller/storagecluster/storagecluster_controller.go:68:1: syntax error: unexpected <<, expecting }```

the e2e test attempts to merge and test thsi PR with the current master branch, which is why that error is showing up.

**Comment by rohantmp** [comment]
Rebased!

**Comment by rohantmp** [comment]
@kshlm CI passed! Can we merge?

**Pull request merged**